app/controllers/reports_controller.rb

```diff
@@ -2,6 +2,54 @@ class ReportsController < ApplicationController
 
   before_action :authenticate_user!
 
+  def new
+    set_form_data
+    @structured_report_data = {}
+
+    @clients = Client.all
+    @projects = Project.all
+    @tasks = Task.all
+    @users = User.all
+  end
+
+  def update
+    set_form_data
+
+    scope_client_ids = @client_ids.empty? ? Client.ids : @client_ids
+    scope_project_ids = @project_ids.empty? ? Project.ids : @project_ids
+    scope_user_ids = @user_ids.empty? ? User.ids : @user_ids
+    scope_task_ids = @task_ids.empty? ? Task.ids : @task_ids
```

Comment:
I think the naming is a bit confusing here. There are 3 different variables for each type (client/project/task/user):
- one which represents the selected checkboxes in the form
- one which represents all selectable options in the form
- one which represents what to show in the report

I don't think the variable names makes this clear.